doc: Adds `retain_backups_enabled`  details on error (#3308)
mongodb/terraform-provider-mongodbatlas: docs/resources/advanced_cluster.md

  Backup uses:
  [Cloud Backups](https://docs.atlas.mongodb.com/backup/cloud-backup/overview/#std-label-backup-cloud-provider) for dedicated clusters.
  [Flex Cluster Backups](https://www.mongodb.com/docs/atlas/backup/cloud-backup/flex-cluster-backup/) for flex clusters.

>   If "`backup_enabled`" : `false`, the cluster doesn't use Atlas backups. This parameter defaults to false.

Reviewer: ```suggestion
  If "`backup_enabled`"  is `false` (default), the cluster doesn't use Atlas backups. 
```
nit

Author: Thanks applied in 818b842ee8354641198a4b24c5dad9305c106b03

> * `retain_backups_enabled` - (Optional) Set to true to retain backup snapshots for the deleted cluster. M10 and above only. This only applies to the `Delete` operation. If you see the `CANNOT_DELETE_SNAPSHOT_WITH_BACKUP_COMPLIANCE_POLICY` error code, set it to explicit `true`.

Reviewer 2: ```suggestion
* `retain_backups_enabled` - (Optional) Set to true to retain backup snapshots upon cluster deletion. M10 and above only. This only applies to the `Delete` operation. If you run into `CANNOT_DELETE_SNAPSHOT_WITH_BACKUP_COMPLIANCE_POLICY` error code during cluster deletion, set it to `true`.
```

Reviewer 2: maybe something like this?
